tiledb/sm/enums/array_type.h

```diff
@@ -74,11 +74,12 @@ inline Status array_type_enum(
   return Status::Ok();
 }
 
-inline Status array_type_is_valid(uint8_t array_type_enum) {
-  if (array_type_enum != 0 && array_type_enum != 1)
-    return Status_Error("Invalid ArrayType enum.");
-
-  return Status::Ok();
+/** Throws error if the input ArrayType's enum is not 0 or 1. */
+inline void ensure_array_type_is_valid(uint8_t array_type_enum) {
+  if (array_type_enum != 0 && array_type_enum != 1) {
```

Comment:
The idea here is to validate against all possible enumeration values. In the `array_type` case, there's only 2 possible enums so it seems less intuitive as `ensure_datatype_is_valid`, which checks enums 0-41

Reply:
I will have to insist here, and I will explain why.
First, `ArrayType::DENSE` and `ArrayType::SPARSE` *are* all possible enumeration values at the moment. It's no different than comparing to `0` and `1`, it's just their named version.
Now, the possible caveat of using unnamed enumeration values like `0` or `1` in such cases is the following: Say in the future we introduce a new array type in our db, say `ArrayType::VERY_SPARSE`. Then as usually we'll grep in our code to find where `ArrayType` is used in order to extend those functions with the right behavior for our new type of array. Since in `ensure_array_type_is_valid` we are not using `ArrayType` at all, it's very possible that we will forget to extend this function without realizing it. And if it happens that test coverage is not enough in this part of the code, we might actually realize it too late. 
In fact it would be even better IMO if we could generalize such checks even more - similar to what `ensure_datatype_is_valid` is doing but without hardcoding - by having a `MAX` value, say `ArrayType::MAX` inour case, defined as the always-last item in all our enums and checking that the our input enum value is less than that MAX value for that type. In that way, just by updating the keeping that MAX value last everytime we add a new type in our enum we wouldn't have to bother changing this type of code at all! So no risk of bug and less work for us. That's out of the scope here, but I just mention it for completeness.